dev/io.openliberty.org.jboss.resteasy.mprestclient.4.0/src/org/jboss/resteasy/microprofile/client/ot/LibertyRestClientBuilderImpl.java

```diff
@@ -0,0 +1,1077 @@
+        }
+
+        builderDelegate.register(new ExceptionMapping(localProviderInstances), 1);
+
+        ClassLoader classLoader = getClassLoader(aClass); // Liberty Change
+
+        T actualClient;
+        ResteasyClient client;
+
+        ResteasyClientBuilder resteasyClientBuilder;
+        if (this.proxyHost != null) {
+            resteasyClientBuilder = builderDelegate.defaultProxy(proxyHost, this.proxyPort);
+        } else {
+            List<String> noProxyHosts = getProxyHostsAsRegex();
+            String envProxyHost = getSystemProperty("http.proxyHost", null);
+            boolean isUriMatched = false;
+            if (envProxyHost != null && !noProxyHosts.isEmpty()) {
+                for (String s : noProxyHosts) {
+                    Pattern p = Pattern.compile(s);
+                    Matcher m = p.matcher(baseURI.getHost());
+                    isUriMatched = m.matches();
+                    if (isUriMatched) {
+                        break;
+                    }
+                }
+            }
+
+            if (envProxyHost != null && !isUriMatched) {
+                // Use proxy, if defined in the env variables
+                resteasyClientBuilder = builderDelegate.defaultProxy(envProxyHost,
+                        Integer.parseInt(getSystemProperty("http.proxyPort", "80")));
+            } else {
+                // Search for proxy settings passed in the client builder, if passed and use them if found
+                String userProxyHost = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_HOST))
+                        .filter(String.class::isInstance).map(String.class::cast).orElse(null);
+
+                Integer userProxyPort = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_PORT))
+                        .filter(Integer.class::isInstance).map(Integer.class::cast).orElse(null);
+
+                String userProxyScheme = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_SCHEME))
+                        .filter(String.class::isInstance).map(String.class::cast).orElse(null);
+
+                if (userProxyHost != null && userProxyPort != null) {
+                    resteasyClientBuilder = builderDelegate.defaultProxy(userProxyHost, userProxyPort, userProxyScheme);
+                } else {
+                    // ProxySelector if applicable
+                    selectHttpProxy().ifPresent(
+                            proxyAddress -> builderDelegate.defaultProxy(proxyAddress.getHostString(), proxyAddress.getPort()));
+
+                    resteasyClientBuilder = builderDelegate;
+                }
+            }
+        }
+
+        if (this.executorService != null) {
+            resteasyClientBuilder.executorService(this.executorService);
+        } else {
+            this.executorService = ContextualExecutors.threadPool();
+            resteasyClientBuilder.executorService(executorService, !executorService.isManaged());
```

Comment:
Have to check if this is right. We are using the RestEasy code and we used other logic in the past. I would hope we have a test for this, but maybe we don't. If we do, it passed if it was in the buckets you modified. This is where OsgiServices used to be used.